src/Hedgehog/Gen.fs

```diff
@@ -162,30 +162,15 @@ module Gen =
     //
 
     /// Generates a random number in the given inclusive range.
-    let inline range (lo : ^a) (hi : ^a) : Gen<'a> =
-        create (Shrink.towards lo) (Random.range lo hi)
+    let inline integral (range : Range<'a>) : Gen<'a> =
+        create (Shrink.towards <| Range.origin range) (Random.integral range)
 
     /// Generates a random number from the whole range of the numeric type.
     let inline bounded () : Gen<'a> =
```

Comment:
Later on, I consider moving `Gen.bounded` to an `Internal` submodule, so it'll be something like `Gen.Internal.bounded` just so that we're in line with the Haskell version for `Gen` and `Range`.